lib/std/Progress.zig

```diff
@@ -810,36 +812,39 @@ fn serialize(serialized_buffer: *Serialized.Buffer) Serialized {
 }
 
 const SavedMetadata = struct {
-    ipc_fd: u16,
+    remaining_read_trash_bytes: u16,
     main_index: u8,
     start_index: u8,
     nodes_len: u8,
+};
 
-    fn getIpcFd(metadata: SavedMetadata) posix.fd_t {
-        return if (is_windows)
-            @ptrFromInt(@as(usize, metadata.ipc_fd) << 2)
-        else
-            metadata.ipc_fd;
+const Fd = enum(i32) {
+    _,
+
+    fn init(fd: posix.fd_t) Fd {
+        return @enumFromInt(if (is_windows) @as(isize, @bitCast(@intFromPtr(fd))) else fd);
     }
 
-    fn setIpcFd(fd: posix.fd_t) u16 {
-        return @intCast(if (is_windows)
-            @shrExact(@intFromPtr(fd), 2)
+    fn get(fd: Fd) posix.fd_t {
+        return if (is_windows)
+            @ptrFromInt(@as(usize, @bitCast(@as(isize, @intFromEnum(fd)))) << 2)
```

Comment:
```suggestion
            @ptrFromInt(@as(usize, @bitCast(@as(isize, @intFromEnum(fd)))))
```
Needs to match `init`, and no need for the extra shift operation when we have 32 bits of storage.